tests/test_matching_algorithms/test_attributed_matching_algorithm.py

```diff
@@ -0,0 +1,143 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of GSy Interface.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+# pylint: disable=missing-function-docstring, protected-access
+import uuid
+from typing import Optional
+
+from pendulum import DateTime
+
+from gsy_framework.data_classes import Offer, BidOfferMatch, Bid
+from gsy_framework.matching_algorithms.attributed_matching_algorithm import (
+    AttributedMatchingAlgorithm)
+
+
+class TestAttributedMatchingAlgorithm:
+    """Tester class for the PreferredPartnersMatchingAlgorithm."""
+    @staticmethod
+    def offer_factory(additional_data: Optional[dict] = None):
+        """Create and return an offer object from default and input values."""
+        additional_data = additional_data or {}
+        return Offer(
+            **{"id": str(uuid.uuid4()),
+               "creation_time": DateTime.now(),
+               "time_slot": DateTime.now(),
+               "price": 10,
+               "energy": 30,
+               "seller": "seller",
+               "seller_id": "seller_id",
+               "seller_origin": "seller",
+               "seller_origin_id": "seller_id",
+               **additional_data})
```

Comment:
it seems that we're using these factories in another test module, can we reuse them somehow?

Reply:
We could even use pytest factories and move them into `conftest.py` (as we already do in gsy-web).
- [conftest.py: sharing fixtures across multiple files](https://docs.pytest.org/en/6.2.x/fixture.html#conftest-py-sharing-fixtures-across-multiple-files)

Reply:
We probably could, however this would introduce dependencies across test classes, that's why I think it is better to have these as static methods outside of all test classes. I will leave it as a post-hackathon TODO and refactor it.